### DISCUSSION TITLE
Path in "APP_DIR" is ignored for web font URIs

### DISCUSSION THREAD

**Discussion by Octo80** [LycheeOrg/Lychee#2623] [Bugs] [answered]
### Lychee version

v6.0.0

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

The path value in "APP_DIR" is not used for the web font URIs and are therefore wrong => results in a HTTP 404 error.

### Steps to reproduce the issue

1. Set the following variable in .env file: APP_DIR=/Lychee/public
2. Access the web frontend and check the developer view "network":
You will see https://www.mydomain.com/build/assets/primeicons-C6QP2o4f.woff2
It should be https://www.mydomain.com/Lychee/public/build/assets/primeicons-C6QP2o4f.woff2

### Diagnostics [REQUIRED]

Errors
info
Have you considered supporting Lychee? :)
Info

Lychee Version (release):                  6.0.0
DB Version:                                6.0.0
composer install:                          --no-dev
APP_ENV:                                   production
APP_DEBUG:                                 false
APP_URL:                                   set
APP_DIR:                                   set
LOG_VIEWER_ENABLED:                        true
VUEJS_ENABLED:                             true
PHOTO_PIPES:                               true
System:                                    Linux
PHP Version:                               8.3.6
PHP User agent:                            Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                  Europe/Berlin
Max uploaded file size:                    500M
Max post size:                             500M
Chunk size:                                12.80 MB
Max execution time:                        200
MySQL Version:                             8.0.39-0ubuntu0.24.04.2
exec() Available:                          yes
Imagick Available:                         1
Imagick Enabled:                           1
Imagick Version:                           1692
GD Version:                                2.3.3
Number of foreign key:                     12 found.

		

Space
Configuration

version:                                   060000
check_for_updates:                         0
sorting_photos_col:                        taken_at
sorting_photos_order:                      ASC
sorting_albums_col:                        max_taken_at
sorting_albums_order:                      ASC
imagick:                                   1
skip_duplicates:                           0
small_max_width:                           0
small_max_height:                          360
medium_max_width:                          1920
medium_max_height:                         1080
lang:                                      en
image_overlay_type:                        desc
default_license:                           none
compression_quality:                       90
grants_full_photo_access:                  1
delete_imported:                           0
mod_frame_enabled:                         1
mod_frame_refresh:                         30
thumb_2x:                                  1
small_2x:                                  1
medium_2x:                                 1
landing_page_enable:                       0
force_migration_in_production:             0
search_public:                             0
hide_version_number:                       0
SL_enable:                                 0
SL_for_admin:                              0
recent_age:                                1
grants_download:                           0
photos_wraparound:                         1
map_display:                               0
zip64:                                     1
map_display_public:                        0
map_provider:                              Wikimedia
force_32bit_ids:                           0
map_include_subalbums:                     0
update_check_every_days:                   3
has_exiftool:                              0
share_button_visible:                      0
import_via_symlink:                        0
has_ffmpeg:                                0
location_decoding:                         0
location_decoding_timeout:                 30
location_show:                             1
location_show_public:                      0
rss_enable:                                0
rss_recent_days:                           7
rss_max_items:                             100
prefer_available_xmp_metadata:             0
editor_enabled:                            1
lossless_optimization:                     0
swipe_tolerance_x:                         150
swipe_tolerance_y:                         250
local_takestamp_video_formats:             .avi|.mov
log_max_num_line:                          1000
unlock_password_photos_with_url_param:     0
nsfw_visible:                              1
nsfw_blur:                                 0
nsfw_warning:                              0
nsfw_warning_admin:                        0
nsfw_banner_override:                      
map_display_direction:                     1
album_subtitle_type:                       oldstyle
upload_processing_limit:                   4
new_photos_notification:                   0
legacy_id_redirection:                     1
zip_deflate_level:                         6
default_album_protection:                  1
album_decoration:                          layers
album_decoration_orientation:              row
allow_username_change:                     1
auto_fix_orientation:                      1
use_job_queues:                            0
random_album_id:                           starred
use_last_modified_date_when_no_exif_date:  0
layout:                                    justified
date_format_photo_thumb:                   M j, Y, g:i:s A e
date_format_photo_overlay:                 M j, Y, g:i:s A e
date_format_sidebar_uploaded:              M j, Y, g:i:s A e
date_format_sidebar_taken_at:              M j, Y, g:i:s A e
date_format_hero_min_max:                  F Y
date_format_hero_created_at:               M j, Y, g:i:s A T
date_format_album_thumb:                   M Y
upload_chunk_size:                         0
nsfw_banner_blur_backdrop:                 0
search_pagination_limit:                   1000
search_minimum_length_required:            4
photo_layout_justified_row_height:         320
photo_layout_masonry_column_width:         300
photo_layout_grid_column_width:            250
photo_layout_square_column_width:          200
photo_layout_gap:                          12
display_thumb_album_overlay:               always
display_thumb_photo_overlay:               hover
default_album_thumb_aspect_ratio:          1/1
use_album_compact_header:                  0
login_button_position:                     left
back_button_enabled:                       0
back_button_text:                          Return to Home
current_job_processing_visible:            1
SA_random_thumbs:                          0
job_processing_queue_visible:              1
login_required:                            0
enable_unsorted:                           1
enable_starred:                            1
enable_recent:                             1
enable_on_this_day:                        1
login_required_root_only:                  1
dark_mode_enabled:                         1
show_keybinding_help_popup:                1
show_keybinding_help_button:               1
disable_se_call_for_actions:               0
enable_se_preview:                         0
slideshow_timeout:                         5
keep_original_untouched:                   1
hide_nsfw_in_smart_albums_and_search:      1

		

### Browser & System [REQUIRED]

Independent because it's not a browser issue (by the way: Firefox 131.0.3, Linux)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment, marked as answer]
It is already fixed on `master` branch. We have not done a release yet.
If you are impatient, you can try the artifact from the build here: https://github.com/LycheeOrg/Lychee/actions/runs/11576424410/artifacts/2118262794
